components/keycloak.key.manager/src/main/java/org/wso2/keycloak/client/KeycloakClient.java

```diff
@@ -327,8 +327,17 @@ public OAuthApplicationInfo buildFromJSON(String s) throws APIManagementExceptio
      */
     @Override
     public OAuthApplicationInfo mapOAuthApplication(OAuthAppRequest oAuthAppRequest) throws APIManagementException {
+        String consumerKey = oAuthAppRequest.getOAuthApplicationInfo().getClientId();
+        String consumerSecret = oAuthAppRequest.getOAuthApplicationInfo().getClientSecret();
 
-        return oAuthAppRequest.getOAuthApplicationInfo();
+        if (StringUtils.isNotBlank(consumerKey) && StringUtils.isNotBlank(consumerSecret)) {
+            OAuthApplicationInfo clientInfo = retrieveApplication(consumerKey);
+            if (clientInfo != null && consumerSecret.equals(clientInfo.getClientSecret())) {
+                return oAuthAppRequest.getOAuthApplicationInfo();
+            }
+        }
+
+        throw new APIManagementException("Invalid credentials");
```

Comment:
Shall we throw more appropriate exception

Reply:
Refactored the code to throw appropriate exceptions